packages/quantic/force-app/main/default/lwc/quanticNoResults/quanticNoResults.js

```diff
@@ -0,0 +1,89 @@
+import { api, LightningElement, track } from 'lwc';
+import { initializeWithHeadless, registerComponentForInit } from 'c/quanticHeadlessLoader';
+
+export default class QuanticNoResults extends LightningElement {
+    /** @type {string}*/
+    @api engineId;
+
+    /** @type {import("coveo").SearchStatus} */
+    searchStatus;
+
+    /** @type {import("coveo").HistoryManager} */
+    historyManager;
+
+    /** @type {import("coveo").QuerySummary} */
+    querySummary;
+
+    /** @type {() => void} */
+    unsubscribeSearchStatus;
+    /** @type {() => void} */
+    unsubscribeHistoryManager;
+    /** @type {() => void} */
+    unsubscribeQuerySummary;
+    /**
+     * @type {boolean}
+     */
+    @api enableCancelLastAction
+
+    /**
+     * @type {boolean}
+     */
+    @api enableSearchTips
+
+    /**
+     * @type {boolean}
+     */
+    @track hasResults;
+
+    /**
+     * @type {number}
+     */
+    @track hasHistory;
+
+    /**
+     * @type {string}
+     */
+    @track query;
+
+    connectedCallback() {
+        registerComponentForInit(this, this.engineId);
+    }
+    
+    renderedCallback() {
+        initializeWithHeadless(this, this.engineId, this.initialize.bind(this));
+    }
+
+    /**
+     * @param {import("coveo").SearchEngine} engine
+     */
+    @api
+    initialize(engine) {
+        this.searchStatus = CoveoHeadless.buildSearchStatus(engine);
+        this.historyManager = CoveoHeadless.buildHistoryManager(engine);
+        this.querySummary = CoveoHeadless.buildQuerySummary(engine);
+        this.unsubscribeSearchStatus = this.searchStatus.subscribe(() => this.updateState());
+        this.unsubscribeHistoryManager = this.historyManager.subscribe(() => this.updateState());
+        this.unsubscribeQuerySummary = this.querySummary.subscribe(() => this.updateState());
+    }
+
+    disconnectedCallback() {
+        if (this.unsubscribeSearchStatus) {
+          this.unsubscribeSearchStatus();
+        }
+        if(this.unsubscribeHistoryManager) {
+            this.unsubscribeHistoryManager();
+        }
+        if(this.unsubscribeQuerySummary) {
+            this.unsubscribeQuerySummary();
+        }
+    }
+    updateState() {
+        this.hasResults = !this.searchStatus.state.firstSearchExecuted ||this.searchStatus.state.isLoading || this.searchStatus.state.hasResults;
```

Comment:
I'm not sure I understand what is going on here.
Should `hasResults` be true if the first search has not been executed?
Should `hasResults` be true if `isLoading` is true?

Reply:
in `atomic-no-results` : 
![image](https://user-images.githubusercontent.com/85293211/127394513-2fd0d00c-d70b-4a75-903d-08722f1ac4b6.png)

I name it `hasResults` for all cases where `quantic-no-results` should not be visible

Reply:
Could we name it something that reflects that to avoid confusion?

Reply:
you know naming is my weakness 😉 , anu suggestion 😄

Reply:
It seems to me that `firstSearchExecuted` and `isLoading` have to be checked to avoid displaying the no results panel before the initial query completes.

In other words, we want the no results panel to show up only when a query has been performed and no results were returned.

Reply:
Perhaps we could rename the property `showNoResultsPanel`?